src/history/hs_rec.c

```diff
@@ -175,10 +175,16 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
         if (!hs_read_all_flag)
             F_CLR(cursor, WT_CURSTD_HS_READ_ALL);
     }
-    if (ret == 0)
+    if (ret == 0) {
```

Comment:
The flag is set at the wrong place. It should be on the previous two search near `__wt_curhs_search_near_after`